File: app/src/main/java/com/kkkcut/e20j/ui/fragment/technical/TechnicalInfoBrandSelectFragment.java
package com.kkkcut.e20j.ui.fragment.technical;

import android.text.Editable;
import android.text.TextUtils;
import android.widget.EditText;
import android.widget.TextView;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.kkkcut.e20j.DbBean.technical.DataManufacturer;
import com.kkkcut.e20j.adapter.TechnicalInfoBradAdapter;
import com.kkkcut.e20j.customView.indexlib.IndexBar.widget.IndexBar;
import com.kkkcut.e20j.customView.indexlib.suspension.SuspensionDecoration;
import com.kkkcut.e20j.dao.KeyInfoDaoManager;
import com.kkkcut.e20j.ui.fragment.BaseBackFragment;
import com.cutting.machine.MachineInfo;
import com.kkkcut.e20j.us.R;

import io.reactivex.rxjava3.core.Observable;
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
import io.reactivex.rxjava3.functions.Consumer;
import io.reactivex.rxjava3.schedulers.Schedulers;
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.Callable;

/* loaded from: classes.dex */
public class TechnicalInfoBrandSelectFragment extends BaseBackFragment {
    public static final String TAG = "TechnicalInfoBrandSelectFragment";

    EditText etSearch;

    IndexBar indexBar;
    private TechnicalInfoBradAdapter mAdapter;
    private SuspensionDecoration mDecoration;

    RecyclerView rvCategoryList;

    TextView tvSideBarHint;
    private List<DataManufacturer> mDatas1 = new ArrayList();
    private List<DataManufacturer> tempData = new ArrayList();

    @Override // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    protected int getContentViewLayoutID() {
        return R.layout.fragment_keyselect_child;
    }

    public static TechnicalInfoBrandSelectFragment newInstance() {
        return new TechnicalInfoBrandSelectFragment();
    }

    @Override // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    protected void initViewsAndEvents() {
        initView();
        getManufacturers();
    }

    private void initView() {
        this.etSearch.setHint(getString(R.string.manufacturer));
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        this.rvCategoryList.setLayoutManager(linearLayoutManager);
        TechnicalInfoBradAdapter technicalInfoBradAdapter = new TechnicalInfoBradAdapter(getContext());
        this.mAdapter = technicalInfoBradAdapter;
        this.rvCategoryList.setAdapter(technicalInfoBradAdapter);
        SuspensionDecoration suspensionDecoration = new SuspensionDecoration(getContext());
        this.mDecoration = suspensionDecoration;
        this.rvCategoryList.addItemDecoration(suspensionDecoration);
        this.rvCategoryList.addItemDecoration(new DividerItemDecoration(getContext(), 1));
        this.indexBar.setmPressedShowTextView(this.tvSideBarHint).setNeedRealIndex(true).setmLayoutManager(linearLayoutManager);
        this.mAdapter.setOnKeySelectItemClickListener(new TechnicalInfoBradAdapter.OnKeySelectItemClickListener() { // from class: com.kkkcut.e20j.ui.fragment.technical.TechnicalInfoBrandSelectFragment$$ExternalSyntheticLambda0
            @Override // com.kkkcut.e20j.adapter.TechnicalInfoBradAdapter.OnKeySelectItemClickListener
            public final void onItemClick(int i) {
                TechnicalInfoBrandSelectFragment.this.m71x9c14395e(i);
            }
        });
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    /* renamed from: lambda$initView$0$com-kkkcut-e20j-ui-fragment-technical-TechnicalInfoBrandSelectFragment, reason: not valid java name */
    public /* synthetic */ void m71x9c14395e(int i) {
        DataManufacturer dataManufacturer = this.tempData.get(i);
        String manufacturerName = dataManufacturer.getManufacturerName();
        if (MachineInfo.isChineseMachine() && !TextUtils.isEmpty(dataManufacturer.getManufacturerNameCN())) {
            manufacturerName = dataManufacturer.getManufacturerNameCN();
        }
        goModel(this.tempData.get(i).getManufacturerId(), manufacturerName);
    }

    private void getManufacturers() {
        addDisposable(Observable.fromCallable(new Callable() { // from class: com.kkkcut.e20j.ui.fragment.technical.TechnicalInfoBrandSelectFragment$$ExternalSyntheticLambda2
            @Override // java.util.concurrent.Callable
            public final Object call() {
                List technicalInfoManufacturers;
                technicalInfoManufacturers = KeyInfoDaoManager.getInstance().getTechnicalInfoManufacturers();
                return technicalInfoManufacturers;
            }
        }).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(new Consumer() { // from class: com.kkkcut.e20j.ui.fragment.technical.TechnicalInfoBrandSelectFragment$$ExternalSyntheticLambda1
            @Override // io.reactivex.functions.Consumer
            public final void accept(Object obj) {
                TechnicalInfoBrandSelectFragment.this.m70x814ceec5((List) obj);
            }
        }));
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    /* renamed from: lambda$getManufacturers$2$com-kkkcut-e20j-ui-fragment-technical-TechnicalInfoBrandSelectFragment, reason: not valid java name */
    public /* synthetic */ void m70x814ceec5(List list) throws Exception {
        this.tempData = list;
        this.mDatas1 = list;
        setData(list);
    }

    public void setData(List<DataManufacturer> list) {
        this.mAdapter.setDatas(list);
        this.indexBar.setmSourceDatas(list).invalidate();
        this.mDecoration.setmDatas(list);
    }

    public void goModel(int i, String str) {
        start(TechnicalInfoModelSelectFragment.newInstance(i, str));
        hideSoftInput();
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    public void afterTextChanged(Editable editable) {
        String manufacturerName;
        this.tempData = new ArrayList();
        for (int i = 0; i < this.mDatas1.size(); i++) {
            DataManufacturer dataManufacturer = this.mDatas1.get(i);
            if (MachineInfo.isChineseMachine()) {
                manufacturerName = dataManufacturer.getManufacturerNameCN();
            } else {
                manufacturerName = dataManufacturer.getManufacturerName();
            }
            if (!TextUtils.isEmpty(manufacturerName) && manufacturerName.toLowerCase().startsWith(editable.toString().toLowerCase())) {
                this.tempData.add(dataManufacturer);
            }
        }
        setData(this.tempData);
    }

    @Override // com.kkkcut.e20j.base.BaseFFragment, com.kkkcut.e20j.androidquick.ui.base.QuickFragment, androidx.fragment.app.Fragment
    public void onDestroyView() {
        super.onDestroyView();
        hideSoftInput();
    }

    @Override // com.kkkcut.e20j.ui.fragment.BaseBackFragment
    public String setTitleStr() {
        return getString(R.string.technical_information);
    }
}

File: app/src/main/java/com/kkkcut/e20j/ui/fragment/technical/TechnicalInfoBrandSelectFragment.kt
package com.kkkcut.e20j.ui.fragment.technical

import android.text.Editable
import android.text.TextUtils
import android.widget.EditText
import android.widget.TextView
import androidx.recyclerview.widget.DividerItemDecoration
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cutting.machine.MachineInfo
import com.kkkcut.e20j.DbBean.technical.DataManufacturer
import com.kkkcut.e20j.adapter.OnKeySelectItemClickListener
import com.kkkcut.e20j.adapter.TechnicalInfoBradAdapter
import com.kkkcut.e20j.customView.indexlib.IndexBar.widget.IndexBar
import com.kkkcut.e20j.customView.indexlib.suspension.SuspensionDecoration
import com.kkkcut.e20j.dao.KeyInfoDaoManager
import com.kkkcut.e20j.ui.fragment.BaseBackFragment
import com.kkkcut.e20j.us.R
import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
import io.reactivex.rxjava3.core.Observable
import io.reactivex.rxjava3.schedulers.Schedulers
import java.util.Locale

/* loaded from: classes.dex */
class TechnicalInfoBrandSelectFragment : BaseBackFragment() {
    var etSearch: EditText? = null

    var indexBar: IndexBar? = null
    private var mAdapter: TechnicalInfoBradAdapter? = null
    private var mDecoration: SuspensionDecoration? = null

    var rvCategoryList: RecyclerView? = null

    var tvSideBarHint: TextView? = null
    private var mDatas = ArrayList<DataManufacturer>()
    private var tempData = ArrayList<DataManufacturer>()

    // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    override fun getContentViewLayoutID(): Int {
        return R.layout.fragment_keyselect_child
    }

    // com.kkkcut.e20j.androidquick.ui.base.QuickFragment
    override fun initViewsAndEvents() {
        initView()
        getManufacturers()
    }

    private fun initView() {
        etSearch!!.hint = getString(R.string.manufacturer)
        val linearLayoutManager = LinearLayoutManager(context)
        rvCategoryList!!.layoutManager = linearLayoutManager
        val technicalInfoBradAdapter = TechnicalInfoBradAdapter(context!!)
        this.mAdapter = technicalInfoBradAdapter
        rvCategoryList!!.adapter = technicalInfoBradAdapter
        val suspensionDecoration = SuspensionDecoration(context)
        this.mDecoration = suspensionDecoration
        rvCategoryList!!.addItemDecoration(suspensionDecoration)
        rvCategoryList!!.addItemDecoration(
            DividerItemDecoration(
                context, 1
            )
        )
        indexBar!!.setmPressedShowTextView(this.tvSideBarHint).setNeedRealIndex(true)
            .setmLayoutManager(linearLayoutManager)
        mAdapter!!.onKeySelectItemClickListener = object : OnKeySelectItemClickListener {
            // from class: com.kkkcut.e20j.ui.fragment.technical.TechnicalInfoBrandSelectFragment$$ExternalSyntheticLambda0
            // com.kkkcut.e20j.adapter.TechnicalInfoBradAdapter.OnKeySelectItemClickListener
            override fun onItemClick(i: Int) {
                this@TechnicalInfoBrandSelectFragment.m71x9c14395e(i)
            }
        }
    }

    /* JADX INFO: Access modifiers changed from: package-private */ /* renamed from: lambda$initView$0$com-kkkcut-e20j-ui-fragment-technical-TechnicalInfoBrandSelectFragment, reason: not valid java name */
    /* synthetic */ fun m71x9c14395e(i: Int) {
        val dataManufacturer = tempData!![i]
        var manufacturerName = dataManufacturer!!.manufacturerName
        if (MachineInfo.isChineseMachine() && !TextUtils.isEmpty(dataManufacturer.manufacturerNameCN)) {
            manufacturerName = dataManufacturer.manufacturerNameCN
        }
        goModel(tempData!![i]!!.manufacturerId, manufacturerName)
    }

    fun getManufacturers() {
            addDisposable(
                Observable.fromCallable {
                    val technicalInfoManufacturers: List<*> =
                        KeyInfoDaoManager.getInstance().technicalInfoManufacturers
                    technicalInfoManufacturers
                }.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(
                { obj ->
                    this@TechnicalInfoBrandSelectFragment.m70x814ceec5(obj as MutableList<*>)
                }, { this.dismissLoadingDialog() })
            )
        }

    /* JADX INFO: Access modifiers changed from: package-private */ /* renamed from: lambda$getManufacturers$2$com-kkkcut-e20j-ui-fragment-technical-TechnicalInfoBrandSelectFragment, reason: not valid java name */
    @Throws(Exception::class)  /* synthetic */ fun m70x814ceec5(list: MutableList<*>?) {
        this.tempData = list as ArrayList<DataManufacturer>
        this.mDatas = list
        setData(list)
    }

    fun setData(list: List<DataManufacturer>) {
        mAdapter!!.setDatas(list)
        indexBar!!.setmSourceDatas(list).invalidate()
        mDecoration!!.setmDatas(list)
    }

    fun goModel(i: Int, str: String?) {
        start(TechnicalInfoModelSelectFragment.newInstance(i, str))
        hideSoftInput()
    }

    /* JADX INFO: Access modifiers changed from: package-private */
    fun afterTextChanged(editable: Editable) {
        var manufacturerName: String
        this.tempData = ArrayList<DataManufacturer>()
        for (i in mDatas!!.indices) {
            val dataManufacturer = mDatas!![i]
            manufacturerName = if (MachineInfo.isChineseMachine()) {
                dataManufacturer!!.manufacturerNameCN
            } else {
                dataManufacturer!!.manufacturerName
            }
            if (!TextUtils.isEmpty(manufacturerName) && manufacturerName.lowercase(Locale.getDefault())
                    .startsWith(
                        editable.toString().lowercase(
                            Locale.getDefault()
                        )
                    )
            ) {
                tempData.add(dataManufacturer)
            }
        }
        setData(this.tempData)
    }

    // com.kkkcut.e20j.base.BaseFFragment, com.kkkcut.e20j.androidquick.ui.base.QuickFragment, androidx.fragment.app.Fragment
    override fun onDestroyView() {
        super.onDestroyView()
        hideSoftInput()
    }

    // com.kkkcut.e20j.ui.fragment.BaseBackFragment
    override fun setTitleStr(): String? {
        return getString(R.string.technical_information)
    }

    companion object {
        const val TAG: String = "TechnicalInfoBrandSelectFragment"

        fun newInstance(): TechnicalInfoBrandSelectFragment {
            return TechnicalInfoBrandSelectFragment()
        }
    }
}
